lib/core/pipes/user-initial.pipe.ts

```diff
@@ -34,15 +34,25 @@ export class InitialUsernamePipe implements PipeTransform {
     transform(user: User, className: string = '', delimiter: string = ''): SafeHtml {
         let safeHtml: SafeHtml = '';
         if (user) {
-            const initialResult = this.getInitialUserName(user.firstName || user.displayName, user.lastName, delimiter);
+            const initialResult = this.getInitialUserName(
+                user.firstName ?? user.displayName,
+                user.lastName,
+                delimiter,
+                user.username
+            );
             safeHtml = this.sanitized.bypassSecurityTrustHtml(`<div id="user-initials-image" class="${className}">${initialResult}</div>`);
         }
         return safeHtml;
     }
 
-    getInitialUserName(firstName: string, lastName: string, delimiter: string): string {
+    getInitialUserName(firstName: string, lastName: string, delimiter: string, username?: string): string {
```

Comment:
I don't like this solution very much. 
Why don't we create a getInitials for unsername and one for firstnameLastname ?

Reply:
another point that is not clear to me is from where this displayName comes from, I know that it was already there... but is very confusing

Reply:
Refactored the code.

We can see now exactly from where we are getting these properties.

Other solution would be just to add:

```
interface User {
    firstName?: string;
    lastName?: string;
    displayName?: string;
    username?: string;
}
```